### PR TITLE
[lte][agw] Fix itti message construction with correct apn size

### PR DESCRIPTION
**Pull request by ulaskozat**
Signed-off-by: Ulas Kozat <kozat@fb.com>

## Summary

GRPC response on apn configuration is not translated correctly onto itti message when the number of returned APN configurations is larger than the maximum allowed by the MME.

## Test Plan

Run full suite of integ tests locally.

Modified s1ap tests.

Output of `sudo journalctl -fu magma@mme` with APNs exceeding the limit:
```
Jul 02 01:41:46 magma-dev-focal mme[1864056]: Received SGW IP Address 0x629000757228
Jul 02 01:41:46 magma-dev-focal mme[1864056]: Received SGW IP Address 0x6290007a2228
Jul 02 01:41:46 magma-dev-focal mme[1864056]: Received SGW IP Address 0x6290007ac228
Jul 02 01:41:46 magma-dev-focal mme[1864056]: [INFO] Sending S6A-AUTHENTICATION_INFORMATION_REQUEST with IMSI: 001010000000001
Jul 02 01:41:46 magma-dev-focal mme[1864056]: [INFO] Received S6A-AUTHENTICATION_INFORMATION_ANSWER for IMSI: 001010000000001; Status: ; StatusCode: 2001
Jul 02 01:41:46 magma-dev-focal mme[1864056]: [DEBUG] Sending S6A-UPDATE_LOCATION_REQUEST with IMSI: 001010000000001
Jul 02 01:41:46 magma-dev-focal mme[1864056]: [INFO] Received S6A-LOCATION-UPDATE_ANSWER for IMSI: 001010000000001; Status: ; StatusCode: 2001
Jul 02 01:41:46 magma-dev-focal mme[1864056]: [WARNING] The number of APNs configured in subscriber data (11) is larger than MME limit of 10. Truncating the list to this MME limit.
Jul 02 01:41:46 magma-dev-focal mme[1864056]: [INFO] sent itti S6A-LOCATION-UPDATE_ANSWER for IMSI: 001010000000001
```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by ardzoht — APPROVED** [review]
lgtm, nice fix!

File: lte/gateway/c/core/oai/lib/s6a_proxy/proto_msg_to_itti_msg.cpp [hunk 1]
```diff
@@ -161,7 +161,6 @@ void convert_proto_msg_to_itti_s6a_update_location_ans(
       SUBSCRIBER_PERIODIC_RAU_TAU_TIMER_VAL;
 
   // apn configuration
-  itti_msg->subscription_data.apn_config_profile.nb_apns = msg.apn_size();
```
Comment (amarpad) [inline lte/gateway/c/core/oai/lib/s6a_proxy/proto_msg_to_itti_msg.cpp:164]: Do we want to log a warning if apn_size is greater than MAX_APN_PER_UE?

**Review by amarpad — APPROVED** [review]

**Comment by themarwhal** [comment]
This is non-trivial to backport to 1.5 due to the directory layout change. @ulaskozat do you mind raising a PR against 1.5 directly?

**Comment by amarpad** [comment]
git cherry-pick should be able to handle directory reparenting, it didn't work for you?

**Comment by amarpad** [comment]
Also if we land it after the changes I am cherry picking it might land cleaner.

**Pull request merged**